prototype_source/ios_gpu_workflow.rst

```diff
@@ -0,0 +1,102 @@
+
+
+Use C++ APIs
+---------------------
+
+In this section, we'll be using the `HelloWorld example <https://github.com/pytorch/ios-demo-app>`_ to demonstrate how to use the C++ APIs. The first thing we need to do is to build a custom LibTorch from Source. Make sure you have deleted the **build** folder from the previous step in PyTorch root directory. Then run the command below
+
+.. code:: shell
+    
+    IOS_ARCH=arm64 ./scripts/build_ios.sh -DUSE_PYTORCH_METAL=ON
+
+Note ``IOS_ARCH`` tells the script to build a arm64 version of Libtorch. This is because in PyTorch, Metal is only available for the iOS devices that support the Apple A9 chip or above. Once the build finished, follow the `Build PyTorch iOS libraries from source <https://pytorch.org/mobile/ios/#build-pytorch-ios-libraries-from-source>`_ section from the iOS tutorial to setup the XCode settings properly. Don't forget to copy the `./mobilenetv2_metal.pt` to your XCode project.
+
+The last step is to change the ``predictImage`` method in ``TorchModule.mm``
+
+.. code:: objective-cd
+    
+    - (NSArray<NSNumber*>*)predictImage:(void*)imageBuffer {
+      torch::jit::GraphOptimizerEnabledGuard opguard(false);
+      at::Tensor tensor = torch::from_blob(imageBuffer, {1, 3, 224, 224}, at::kFloat).metal();
+      auto outputTensor = _impl.forward({tensor}).toTensor().cpu();
+      ...
+      return nil;
+    }
+
+As you can see, we simply just call ``.metal()`` to move our input tensor from CPU to GPU, and then call ``.cpu()`` to move the result back. Internally, ``.metal()`` will copy the input data from the CPU buffer to a GPU buffer and convert the memory format. When `.cpu()` is invoked, the GPU command buffer will be flushed and synced. The final result will then be copied back from the GPU buffer back to a CPU buffer.
+
+If everything works fine, you should see the inference results being showed on the screen. The result below is captured from a iPhone11 device
+
+.. code:: shell
+
+    - timber wolf, grey wolf, gray wolf, Canis lupus
+    - malamute, malemute, Alaskan malamute
+    - Eskimo dog, husky
+
+You may notice that the results are slighly different from the `results <https://pytorch.org/mobile/ios/#install-libtorch-via-cocoapods>`_ we got from the CPU model as shown in the iOS tutorial. This because by default Metal uses fp16 rather than fp32 to compute. The precision loss from the result is expected.
+
+
+Conclusion
+----------
+
+In this tutorial, we demonstrated how to convert the existing torchscript model to a GPU compatible model for iOS. We walked through the HelloWorld example to show how to use the C++ APIs. Please be aware of that GPU feature is still under development, new operators will continue be added. APIs are subject to change in the future versions.
+
+Thanks for reading! As always, we welcome any feedback, so please create an issue `here <https://github.com/pytorch/pytorch/issues>`_ if you have any.
+
```

Comment:
Looks good! Only comment would be are there any related APIs docs or models (e.g. [Mobile Net v2](https://pytorch.org/hub/pytorch_vision_mobilenet_v2/)) that can be linked either in a "Learn More" section at the end or in the introduction?